### ISSUE TITLE
[Bug Report] No method exposed to set cert chain when using X509 auth

### ISSUE THREAD

**Issue by ancaantochi** [Azure/azure-iot-sdk-csharp#1566]
# Context

- **OS, version, SKU and CPU architecture used:** All
- **Application's .NET Target Framework :** All
- **Device:** All
- **SDK version used:** 

## Description of the issue
If setting up device auth with CA and creating a chain which is more than 2 certs (like Root CA -> Intermediate Cert -> Idenity Cert) it's not possible to set in the SDK the chain and authentication fails because the chain was not specified.

Workaround is to install the chain on the device.

## Code sample exhibiting the issue
```
var cert = new X509Certificate2(@"<absolute-path-to-your-device-pfx-file>", "1234"); 
var auth = new DeviceAuthenticationWithX509Certificate("<device-id>", cert);
```

Authentication will fail when device cert was signed by an intermediate.

**Comment by abhipsaMisra** [comment]
This does look like a bug on the SDK, we'll fix this.

**Comment by abhipsaMisra** [comment]
The fix (#1742) has now been released: https://github.com/Azure/azure-iot-sdk-csharp/releases/tag/2021-01-21
